### DISCUSSION TITLE
myloader "Thread -1" when -t 2

### DISCUSSION THREAD

**Discussion by astesin** [mydumper/mydumper#1028] [Q&A] [answered]
While trying to reduce active threads from default 4 to 2, I've set -v 3 -t 2 and got the following in the log:

** Message: 23:16:31.851: Thread 1: restoring `some`.`table` part 1 of 1 from some.table.00000.sql. Progress 7 of 745. Using 1 of **4 threads**.
** Message: 23:16:31.852: Thread **-1**: Enqueuing index for table: `some`.`othertable`

Now scratching my head, why still 4 threads in the 1st message, and what does "thread -1" mean...
_myloader0.13.1-2, built against MySQL 5.7.40-43 with SSL support with GZIP_

**Comment by davidducos** [comment, marked as answer]
Hi @astesin 
I will fix this:
```
      dbt->max_threads=max_threads_per_table;
```
 with this:
```
      dbt->max_threads=max_threads_per_table>num_threads?num_threads:max_threads_per_table;
```
To avoid showing the wrong number of limit. Anyways, I think that the number of threads used is not correct but it is using the number that you configure.

**Reply by davidducos** [reply]
Done

**Comment by astesin** [comment]
Thanks a lot! For a newbie like me, those strange numbers are somehow
frightening because how do I know that it is my fault or a software issue?
